src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractWalletController.cs

```diff
@@ -0,0 +1,337 @@
+            }
+        }
+
+        [Route("send-transaction")]
+        [HttpPost]
+        public IActionResult SendTransaction([FromBody] SendTransactionRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+                return ModelStateErrors.BuildErrorResponse(this.ModelState);
+
+            if (!this.connectionManager.ConnectedPeers.Any())
+                throw new WalletException("Can't send transaction: sending transaction requires at least one connection!");
+
+            try
+            {
+                Transaction transaction = this.network.CreateTransaction(request.Hex);
+
+                var model = new WalletSendTransactionModel
+                {
+                    TransactionId = transaction.GetHash(),
+                    Outputs = new List<TransactionOutputModel>()
+                };
+
+                foreach (TxOut output in transaction.Outputs)
+                {
+                    bool isUnspendable = output.ScriptPubKey.IsUnspendable;
+
+                    string address = GetAddressFromScriptPubKey(output);
+                    model.Outputs.Add(new TransactionOutputModel
+                    {
+                        Address = address,
+                        Amount = output.Value,
+                        OpReturnData = isUnspendable ? Encoding.UTF8.GetString(output.ScriptPubKey.ToOps().Last().PushData) : null
+                    });
+                }
+
+                this.broadcasterManager.BroadcastTransactionAsync(transaction).GetAwaiter().GetResult();
+
+                TransactionBroadcastEntry transactionBroadCastEntry = this.broadcasterManager.GetTransaction(transaction.GetHash());
+                if (!string.IsNullOrEmpty(transactionBroadCastEntry?.ErrorMessage))
+                {
+                    this.logger.LogError("Exception occurred: {0}", transactionBroadCastEntry.ErrorMessage);
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, transactionBroadCastEntry.ErrorMessage, "Transaction Exception");
+                }
+
+                return this.Json(model);
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a string that represents the receiving address for an output.For smart contract transactions,
+        /// returns the opcode that was sent i.e.OP_CALL or OP_CREATE
+        /// </summary>
+        private string GetAddressFromScriptPubKey(TxOut output)
+        {
+            if (output.ScriptPubKey.IsSmartContractExec())
+                return output.ScriptPubKey.ToOps().First().Code.ToString();
```

Comment:
Unclear to me why we return the opcode here instead of the destination address if OP_CALL or nothing if OP_CREATE.